compiler/src/dotty/tools/dotc/core/tasty/TastyBuffer.scala

```diff
@@ -48,6 +48,10 @@ class TastyBuffer(initialSize: Int) {
 
   // -- Output routines --------------------------------------------
 
+  /** Write a boolean value. */
+  def writeBoolean(b: Boolean): Unit =
```

Comment:
I would write the byte explicitly `pickleComment` and read the byte in `CommentUnpickler`. This abstraction is only used once and hides the underlying representation. It would make it too easy to write two calls to `writeBoolean` instead of one to `writeByte` (with bitset representation).